File: android/core/observability/src/main/java/com/enterprise/core/observability/FirebaseCrashReporter.kt
package com.enterprise.core.observability

import com.enterprise.core.domain.interfaces.CrashReporter
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Firebase Crashlytics implementation of CrashReporter.
 * No SDK bleed: Crashlytics API stays here, never leaks to Domain/Presentation.
 */
@Singleton
class FirebaseCrashReporter @Inject constructor() : CrashReporter {

    override fun logNonFatal(exception: Exception) {
        // Firebase.crashlytics.recordException(exception)
        // Placeholder: actual Firebase SDK call in production
        android.util.Log.e(TAG, "Non-fatal: ${exception.message}", exception)
    }

    override fun setCustomKey(key: String, value: String) {
        // Firebase.crashlytics.setCustomKey(key, value)
        android.util.Log.d(TAG, "Custom key: $key = $value")
    }

    override fun logBreadcrumb(message: String) {
        // Firebase.crashlytics.log(message)
        android.util.Log.d(TAG, "Breadcrumb: $message")
    }

    companion object {
        private const val TAG = "CrashReporter"
    }
}

File: ios/EnterpriseMobileApp/Sources/Core/Observability/FirebaseCrashReporter.swift
import Foundation
import CoreDomain

/// Firebase Crashlytics implementation - SDK stays in Data/Infrastructure layer only.
public final class FirebaseCrashReporter: CrashReporter, @unchecked Sendable {
    public init() {}

    public func logNonFatal(_ error: Error) {
        // In production: Crashlytics.crashlytics().record(error: error)
        print("[CrashReporter] Non-fatal: \(error.localizedDescription)")
    }

    public func setCustomKey(_ key: String, value: String) {
        // In production: Crashlytics.crashlytics().setCustomValue(value, forKey: key)
        print("[CrashReporter] Key: \(key) = \(value)")
    }

    public func logBreadcrumb(_ message: String) {
        // In production: Crashlytics.crashlytics().log(message)
        print("[CrashReporter] Breadcrumb: \(message)")
    }
}
